freedomofpress/securedrop
# Design updates for SecureDrop 1.0.0, rev1

Updates colors throughout to a new "SecureDrop Blue" and a darker
"Grimace Blue", a new "danger" color, and blue gradients on the
index page; updates the favicon and the default logo.

Resolves #4500
Resolves #4501
Resolves #4502
Resolves #4503

Additional tweaks chiefly agreed upon with Nina Alter (@ninavizz)
during pair development:

Global:
- Removal of some button icons that did not disambiguate meaning
- Tweaks to default button letter spacing, font weight
- Pick headline color that fits with new overall theme
- Ensure consistent use of headline levels
- Removed unused colors, styles, icons
- More careful spacing between icons and text on buttons

Source Interface:
- New trashcan icon for deleting replies
- Flipped order of introductory text on submit page by importance

Journalist Interface:
- Restyled timestamps, "0 docs, 1 message" type counters to be less easily confused with buttons
- Restyled "Select all" type buttons to not have checkbox icons on them, to ensure they're not confused with actual checkboxes
- Added spacing between Select buttons and other actions, to make logical distinction more obvious
- Removed download icon from each table row, as it did not convey sufficient meaning to justify keeping it
- Removed "Reply" icon from Reply headline, and left-aligned it, for consistency with other headlines
- Made textbox the same width as the table above it, and removed fixed column width, to ensure proper scaling across viewport sizes
- Right-aligned "Submit" button consistent with other forms
- Removed hover from "X" of modals, as it did not really add value here

Admin Interface:
- Made it more obvious when the Delete icon is disabled

Out of scope for this rev:
- String changes
- Typography changes
- Docs screenshot updates (will be done after final rev)

## Test plan

- Check the changes against the changelog above.
- Verify all paths through the Source Interface, including:
   - submitting a file or message
   - changing the UI language
   - deleting a reply from a journalist
- Verify all paths through the Journalist Interface, including:
   - "Flag for reply" flow
   - replying to a source
   - deletion of collections and sources
   - deletion of replies by sources
   - changing codename of a source
   - adding a user
   - other administrative functions

When stepping through these paths, make note of any obvious design regressions, or design changes which you are uncertain about ("is it intentional that the hover behavior between these two buttons is inconsistent", etc.).

## Deployment considerations

- Screenshots in docs will have to be updated after final rev
- We'll reach out to news organizations prior to 1.0.0 to keep them informed about the logo change, so they can account for it on landing pages and in other materials
- We'll prominently explain the UI changes in the changelog for 1.0.0

## Screenshots

![Screen Shot 2019-07-30 at 17 35 48-fullpage](https://user-images.githubusercontent.com/213636/62175020-f000f300-b2f0-11e9-83b4-066fe07104ca.png)
![Screen Shot 2019-07-30 at 17 35 50-fullpage](https://user-images.githubusercontent.com/213636/62175021-f000f300-b2f0-11e9-9fc1-ad094e972f05.png)
![Screen Shot 2019-07-30 at 17 36 13-fullpage](https://user-images.githubusercontent.com/213636/62175022-f000f300-b2f0-11e9-939e-5aef68d37efc.png)
![Screen Shot 2019-07-30 at 17 36 26-fullpage](https://user-images.githubusercontent.com/213636/62175023-f000f300-b2f0-11e9-9978-2304fee0228e.png)
![Screen Shot 2019-07-30 at 17 36 33-fullpage](https://user-images.githubusercontent.com/213636/62175024-f0998980-b2f0-11e9-90b6-799eb5cd7b8b.png)
![Screen Shot 2019-07-30 at 17 37 06-fullpage](https://user-images.githubusercontent.com/213636/62175026-f0998980-b2f0-11e9-9459-1991cda945f3.png)
![Screen Shot 2019-07-30 at 17 37 14-fullpage](https://user-images.githubusercontent.com/213636/62175028-f0998980-b2f0-11e9-8fe2-2ac6671d9330.png)
![Screen Shot 2019-07-30 at 17 37 49-fullpage](https://user-images.githubusercontent.com/213636/62175029-f0998980-b2f0-11e9-91b8-1cb4912daf5c.png)
![Screen Shot 2019-07-30 at 17 38 06-fullpage](https://user-images.githubusercontent.com/213636/62175030-f1322000-b2f0-11e9-84fe-694e7210a159.png)
